pytest/tests/sanity/state_sync_decentralized.py

```diff
@@ -47,58 +46,42 @@
-        (node_config_dump,
-         node_config_sync) = state_sync_lib.get_state_sync_configs_pair(
-             tracked_shards=None)
-
-        # State snapshot is disabled for dumper. We only want to dump the headers.
-        node_config_dump[
-            "store.state_snapshot_config.state_snapshot_type"] = "Disabled"
-
-        # State snapshot is enabled for validators. They will share parts of the state.
-        node_config_sync[
-            "store.state_snapshot_config.state_snapshot_type"] = "Enabled"
-        node_config_sync["tracked_shards_config"] = "NoShards"
+    def _prepare_cluster(self):
+        node_config = state_sync_lib.get_state_sync_config_p2p(
+            tracked_shards_config="NoShards")
 
         # Validators
-        configs = {x: node_config_sync.copy() for x in range(NUM_VALIDATORS)}
-
-        # Dumper
-        configs[NUM_VALIDATORS] = node_config_dump
-
-        if with_rpc:
-            # RPC
-            configs[NUM_VALIDATORS + 1] = node_config_sync.copy()
-            # RPC tracks all shards.
-            configs[NUM_VALIDATORS + 1]["tracked_shards_config"] = "AllShards"
-            # RPC node does not participate in state parts distribution.
-            configs[NUM_VALIDATORS + 1][
-                "store.state_snapshot_config.state_snapshot_type"] = "Disabled"
+        configs = {x: node_config.copy() for x in range(NUM_VALIDATORS)}
+
+        # RPC
+        configs[NUM_VALIDATORS] = node_config.copy()
+        # RPC tracks all shards.
+        configs[NUM_VALIDATORS]["tracked_shards_config"] = "AllShards"
+        # RPC node does not serve state sync snapshots.
+        configs[NUM_VALIDATORS][
+            "store.state_snapshot_config.state_snapshot_type"] = "Disabled"
 
         nodes = start_cluster(
             num_nodes=NUM_VALIDATORS,
-            num_observers=1 + (1 if with_rpc else 0),
+            num_observers=1,
             num_shards=NUM_SHARDS,
             config=None,
             genesis_config_changes=[
-                ["epoch_length", EPOCH_LENGTH], ["shard_layout", SHARD_LAYOUT],
-                [
-                    "shuffle_shard_assignment_for_chunk_producers",
-                    shuffle_shard_assignment
-                ], ["block_producer_kickout_threshold", 0],
+                # Rotate tracked shards to trigger state sync
+                ["shuffle_shard_assignment_for_chunk_producers", True],
+                ["epoch_length", EPOCH_LENGTH],
+                ["shard_layout", SHARD_LAYOUT],
+                ["block_producer_kickout_threshold", 0],
                 ["chunk_producer_kickout_threshold", 0]
             ],
             client_config_changes=configs)
 
         for node in nodes:
             node.stop_checking_store()
 
-        self.dumper_node = nodes[NUM_VALIDATORS]
-        self.rpc_node = nodes[NUM_VALIDATORS +
-                              1] if with_rpc else self.dumper_node
```

Comment:
heh why did we have a dumper in tests for decentralized state sync ;)

Reply:
I think the idea was that since we had decentralization done for the parts and not the headers, the dumper would serve only the headers to allow this test to work.